src/receive_imf/tests.rs

```diff
@@ -4892,19 +4867,13 @@ async fn test_leave_protected_group_missing_member_key() -> Result<()> {
             ("b@b", "bob@example.net"),
         )
         .await?;
+
+    // We fail to send the message.
     assert!(remove_contact_from_chat(alice, group_id, ContactId::SELF)
         .await
         .is_err());
-    assert!(is_contact_in_chat(alice, group_id, ContactId::SELF).await?);
-    alice
-        .sql
-        .execute(
-            "UPDATE acpeerstates SET addr=? WHERE addr=?",
-            ("bob@example.net", "b@b"),
-        )
-        .await?;
-    remove_contact_from_chat(alice, group_id, ContactId::SELF).await?;
-    alice.pop_sent_msg().await;
+
+    // The contact is already removed anyway.
     assert!(!is_contact_in_chat(alice, group_id, ContactId::SELF).await?);
```

Comment:
What happens with messages that we continue to receive? They should be visible so that the user can try to solve the problem somehow, e.g. asking somebody to remove them effectively.

Also if we're not in the group anymore, an appropriate info message should be present, i think this should be checked.

Reply:
> Current workaround is to PM someone and ask to remove you again if this happens.

I think this is the best we can do. We shouldn't hide problems with message delivery from users and also it's better to avoid sending any automatic messages, this was discussed a lot already.

> An info message "you are not part of the group but keep receiving messages" will be just a noise in this case.

No, i just meant that even if the self-removal message failed to be sent, there should be an info message that the user left the group, but AFAIU it already works this way.